### DISCUSSION TITLE
windows uv_poll_init

### DISCUSSION THREAD

**Discussion by andyx666888** [libuv/libuv#4385] [Q&A] [answered]
i wanna make ping with libuv 
    https://github.com/joyent/libuv/issues/777
so i use uv_poll_t, 
   SOCKET sock = socket(AF_INET, SOCK_RAW, IPPROTO_ICMP);
    uv_poll_init(loop, &icmpPollHandle, sock);
BUT
   https://github.com/libuv/libuv/blob/v1.x/src/win/poll.c#L411
   uv__get_osfhandle always return -1, i change i to 
  return uv_poll_init_socket(loop, handle, (SOCKET) fd);
  it can work and very well
so am i wrong ?
thanks

* **Version**: 1.48
* **Platform**: windows

**Comment by bnoordhuis** [comment, marked as answer]
uv_poll_init_socket is what you should be using, yes.

**Reply by andyx666888** [reply]
thank you very much and very much
